src/OpenTelemetry.AutoInstrumentation/Logging/FileSink.cs

```diff
@@ -0,0 +1,78 @@
+﻿//------------------------------------------------------------------------------
+// <auto-generated />
+// This comment is here to prevent StyleCop from analyzing a file originally from Serilog.
+//------------------------------------------------------------------------------
+
+// Copyright 2013-2019 Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using System.Text;
+using OpenTelemetry.AutoInstrumentation.Logging;
+
+namespace OpenTelemetry.AutoInstrumentation.Logging
+{
+    internal sealed class FileSink : ISink, IDisposable
+    {
+        readonly TextWriter _output;
+        readonly FileStream _underlyingStream;
+        readonly WriteCountingStream _countingStreamWrapper;
+        readonly object _syncRoot = new object();
+        static readonly long FileSizeLimitBytes = 10 * 1024 * 1024;
```

Comment:
I suggest starting with unbounded file writting
especially that currently the user would not be able to find out the file reached its limit and it is also not configurable
we can enhance (e.g. add file rotation) in next PRs

Reply:
I assumed having file growth bounded is preferred to unbounded file growth. I implemented it this way and didn't make if configurable because I wanted to keep it simple for now (and assumed we want to add file rotation in the near future).